reframe/REFRAME_KNOWN_METHODS.md

```diff
@@ -203,6 +208,39 @@ Response:
 }}
 ```
 
+### FindProviders Blinded
+
+A message for finding nodes with interest in a given key using double hashing to blind the key being requested.
+
+```ipldsch
+    type FindHashedSHA256Request struct {
+        Query Bytes
+    }
+```
+
+The query is a derived hash of the multihash being requested.
+It is constructed by taking the raw bytes of the multihash, prepending the ascii bytes "CR_DOUBLEHASH", and taking the SHA256 hash of that data.
+The resulting digest is then packed itself into a multihash, using the multihash code identifier multihash.DBL_SHA2_256.

```

Comment:
What is the contingency for replacing SHA256 with other function in the future?
Creating a new message type?

Would it be acceptable renaming these requests to `FingHashedProvidersRequest` and move hash function multicodec to struct field?

Reply:
The current contingency is a new message type.

having it in the request is sub-optimal because it would mean another round of probing for what types of hashes are supported.

changing the hash algorithm will be 'a big deal' and something we don't do often, so using the existing 'identify' layer to identify which one the server supports seems preferable to the client trying other ones and getting told they're not supported.

we expect handlers of FindProviders to support 1 or at most 2 hash algorithms at any given time.